data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OrderedConsumerVerticle.java

```diff
@@ -163,12 +164,32 @@ private synchronized OrderedAsyncExecutor executorFor(final TopicPartition topic
     return executor;
   }
 
-  private boolean isWaitingForTasks() {
-    for (OrderedAsyncExecutor value : this.recordDispatcherExecutors.values()) {
-      if (value.isWaitingForTasks()) {
-        return true;
+  private boolean areAllExecutorsBusy() {
+    if (recordDispatcherExecutors.isEmpty()) {
+      // No executors
+      return false;
+    }
+
+    var res = true;
+    final var toResume = new HashSet<TopicPartition>();
+    final var toPause = new HashSet<TopicPartition>();
+
+    for (var executor : recordDispatcherExecutors.entrySet()) {
+      if (executor.getValue().isWaitingForTasks()) {
+        toResume.add(executor.getKey());
+        res = false;
+      } else {
+        toPause.add(executor.getKey());
       }
     }
-    return this.recordDispatcherExecutors.size() == 0;
+
+    if (!toPause.isEmpty()) {
+      this.consumer.pause(toPause);
+    }
+    if (!toResume.isEmpty()) {
+      this.consumer.resume(toResume);
+    }
```

Comment:
Didn't Clement mention something around pause/replay issues on vertx Kafka, @pierDipi ?

Reply:
Yes, but this happens only for some calls https://github.com/vert-x3/vertx-kafka-client/issues/53